src/app/app.component.ts

```diff
@@ -135,6 +133,13 @@ export class AppComponent implements OnInit, AfterViewInit {
     this.user = this.userService.user;
     this.notifications = this.notificationService.getNotifications();
     this.showClose = true;
+
+    const apiEndpoint = this.config.getSettings().apiEndpoint;
+    const accessToken = this.oauthService.getAccessToken();
```

Comment:
Wonder, do we need a `loading` attribute on the controller and then use the `syndesis-loading` component in the view so that the setup page doesn't flash up until the `isSetupPending()` call is fired?

Reply:
That would be convenient. I'll see if I can take a look now.

Reply:
This would be easier to do with observables than with promises (which we are currently using). I think it's an enhancement that can be done with TP2. Wdyt?

Reply:
Can do it without observables as well, `syndesis-loading` just needs a boolean value.  From a promise you may need to inject `ChangeDetectorRef` and manually trigger change detection.